standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -870,44 +870,88 @@ private boolean isViewTable(String catName, String dbName, String tblName) throw
     String orderForFilter = (max != null) ? " order by \"PART_NAME\" asc" : "";
 
     String queryText =
-        "select " + PARTITIONS + ".\"PART_ID\" from " + PARTITIONS + ""
-      + "  inner join " + TBLS + " on " + PARTITIONS + ".\"TBL_ID\" = " + TBLS + ".\"TBL_ID\" "
-      + "    and " + TBLS + ".\"TBL_NAME\" = ? "
-      + "  inner join " + DBS + " on " + TBLS + ".\"DB_ID\" = " + DBS + ".\"DB_ID\" "
-      + "     and " + DBS + ".\"NAME\" = ? "
-      + join(joinsForFilter, ' ')
-      + " where " + DBS + ".\"CTLG_NAME\" = ? "
-      + (StringUtils.isBlank(sqlFilter) ? "" : (" and " + sqlFilter)) + orderForFilter;
-    Object[] params = new Object[paramsForFilter.size() + 3];
-    params[0] = tblNameLcase;
-    params[1] = dbNameLcase;
-    params[2] = catNameLcase;
-    for (int i = 0; i < paramsForFilter.size(); ++i) {
-      params[i + 3] = paramsForFilter.get(i);
-    }
+            "select " + PARTITIONS + ".\"PART_ID\" from " + PARTITIONS + ""
+                    + "  inner join " + TBLS + " on " + PARTITIONS + ".\"TBL_ID\" = " + TBLS + ".\"TBL_ID\" "
+                    + "    and " + TBLS + ".\"TBL_NAME\" = ? "
+                    + "  inner join " + DBS + " on " + TBLS + ".\"DB_ID\" = " + DBS + ".\"DB_ID\" "
+                    + "     and " + DBS + ".\"NAME\" = ? "
+                    + join(joinsForFilter, ' ')
+                    + " where " + DBS + ".\"CTLG_NAME\" = ? "
+                    + (StringUtils.isBlank(sqlFilter) ? "" : (" and " + sqlFilter)) + orderForFilter;
 
-    long start = doTrace ? System.nanoTime() : 0;
-    Query query = pm.newQuery("javax.jdo.query.SQL", queryText);
-    List<Object> sqlResult = executeWithArray(query, params, queryText, ((max == null)  ? -1 : max.intValue()));
-    long queryTime = doTrace ? System.nanoTime() : 0;
-    MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
-    if (sqlResult.isEmpty()) {
-      return Collections.emptyList(); // no partitions, bail early.
+    int parametersToProcess = paramsForFilter.size();
+
+    List<Long> result = new ArrayList<>();
+
+    // If this method has been called by default we need to query the tables once. Hence set the minimum numbner of
+    // iterations to 1.
+    int iterations = 1;
+
+
+    // The first three parameters in the query are the table, database and the catalog names. This leaves us
+    // JDBC_STMT_PARAM_LIMIT - 3 parameters that can be passed to the underlying JDBC driver. If the number of
+    // input parameters is greater than this limit, we will have to process them in multiple iterations.
+    if (parametersToProcess > (JDBC_STMT_PARAM_LIMIT - 3)) {
+      // Find the number of iterations required from the number of input parameters and sets of 32764 parameters
+      // that can be created from it.
+      iterations = parametersToProcess / (JDBC_STMT_PARAM_LIMIT - 3);
+      iterations += ((parametersToProcess % (JDBC_STMT_PARAM_LIMIT - 3) != 0) ? 1 : 0);
```

Comment:
we should be able to merge these 2 statements. iterations is initialized to 1. Within this conditional block, if we just added the result of integer division to the value of iterations, should be good right? This code is only executed when the params is greater than the limit.

Reply:
I used his code to distinguish the case when the number of parameters is a multiple of JDBC_STMT_PARAM_LIMIT - 3, vs when it is not.

For example,

when parametersToProcess = 32764 * 2 => iterations = 2

However 

when parametersToProcess = 32764 * 2 + 11 => iterations = 3

32764 parameters in the first iteration
32764 parameters in the second iteration
11 parameters in the third iteration